libs/facilities/frontend/view/src/lib/components/detail/detail.page.ts

```diff
@@ -1,84 +1,205 @@
+        { from: this._28MinutesAgo, to: this._currentTime },
+    ));
+
+    protected theme = convertThemeName(themeSwitcher.getCurrentTheme());
+    protected readonly defaultOptions: EChartsOption = {
+        xAxis: {
+            type: 'time',
+            name: 'Time',
+            nameLocation: 'middle',
+            nameGap: 30,
+        },
+        yAxis: {
+            type: 'value',
+            name: 'Value',
+            nameLocation: 'middle',
+            nameGap: 30,
+        },
+        legend: {
+            top: 30,
+            left: 20,
+            right: 20,
+        },
+        grid: {
+            top: 80,
+        },
+    };
+
+    protected readonly pumpChart: Signal<EChartsOption | undefined> = computed(() => {
+        const pumpData = this.pumpData();
+        if (!pumpData)
+            return undefined;
+
+        const Flow = pumpData.map((item) => [ item.time, item['Flow'] ]);
+        const MotorCurrent = pumpData.map((item) => [ item.time, item['MotorCurrent'] ]);
+        const StuffingBoxTemperature = pumpData.map((item) => [ item.time, item['StuffingBoxTemperature'] ]);
+        const PressureIn = pumpData.map((item) => [ item.time, item['PressureIn'] ]);
+        const PressureOut = pumpData.map((item) => [ item.time, item['PressureOut'] ]);
+
+        return {
+            ...this.defaultOptions,
+            title: {
+                text: 'Pump Data',
+                left: 'center',
+            },
+            series: [
+                {
+                    name: 'Flow (l/s)',
+                    type: 'line',
+                    data: Flow,
+                    itemStyle: { color: Colors.WATER },
+                    lineStyle: { color: Colors.WATER },
+                },
+                {
+                    name: 'Motor Current (V)',
+                    type: 'line',
+                    itemStyle: { color: Colors.MOTORCURRENT },
+                    lineStyle: { color: Colors.MOTORCURRENT },
+                    data: MotorCurrent,
+                },
+                {
+                    name: 'Stuffing Box Temperature (°C)',
+                    type: 'line',
+                    itemStyle: { color: Colors.TEMPERATURE },
+                    lineStyle: { color: Colors.TEMPERATURE },
+                    data: StuffingBoxTemperature,
+                },
+                {
+                    name: 'Pressure In (hPa)',
+                    type: 'line',
+                    itemStyle: { color: Colors.PRESSURE1 },
+                    lineStyle: { color: Colors.PRESSURE1 },
+                    data: PressureIn,
+                },
+                {
+                    name: 'Pressure Out (hPa)',
+                    type: 'line',
+                    itemStyle: { color: Colors.PRESSURE2 },
+                    lineStyle: { color: Colors.PRESSURE2 },
+                    data: PressureOut,
+                },
+            ],
```

Comment:
It is there because the data is set. I understand that it is not a good solution so I put the other things outside.
The thing is, now I have to set the data like this:
```
pumpChart.series[i].data = ...
```
which is also not very intuitive, but still probably a better solution

Reply:
Why dont split it up?

```ts
const staticPumpData =   private readonly pumpOptions: EChartsOption = {
        title: {
            text: 'Pump Data',
            left: 'center',
        },
        series: [
            {
                name: 'Flow (l/s)',
                type: 'line',
                itemStyle: { color: Colors.WATER },
                lineStyle: { color: Colors.WATER },
            },
            {
                name: 'Motor Current (V)',
                type: 'line',
                itemStyle: { color: Colors.MOTORCURRENT },
                lineStyle: { color: Colors.MOTORCURRENT },
            },
            {
                name: 'Stuffing Box Temperature (°C)',
                type: 'line',
                itemStyle: { color: Colors.TEMPERATURE },
                lineStyle: { color: Colors.TEMPERATURE },
            },
            {
                name: 'Pressure In (hPa)',
                type: 'line',
                itemStyle: { color: Colors.PRESSURE1 },
                lineStyle: { color: Colors.PRESSURE1 },
            },
            {
                name: 'Pressure Out (hPa)',
                type: 'line',
                itemStyle: { color: Colors.PRESSURE2 },
                lineStyle: { color: Colors.PRESSURE2 },
            },
        ],
    };
}

....
export class DeatilPage 
....
 private readonly pumpOptions: EChartsOption = {
        ...this.defaultOptions,
        ...staticPumpData
}
```

Reply:
I think that is exactly what I did if you go to the new changes (name is pumpOptions for static and then pumpChart for the actual Signal). If you want it to be in a const.ts file in the same directory I can add that. The thing is to show the actual data we still need to update the series[i].data attributes. That is done in the computed() part

Reply:
Ah I didnt see thats github :(